File: Model/Loop.mjs
import {Macro} from "./Macro.mjs"
import {Condition} from "./Condition.mjs"
class Loop extends Macro{
    constructor(condition){
        super()
        this.classCode = Macro.TYPE.LOOP
        this.condition = condition
        this.condition.value.counter = this
        this.counter = 0
        this.childMacros = []
    }
    child(){
        return this.childMacros
    }

    label(){
        return  'Repeat until ' + this.condition.label()
    }

    count(){
        return this.counter
    }
}

export {
    Loop
}

File: Controller/Executor/Model/Loop.py
from Model.Macro import Macro
from typing import List
from Model.Condition import Condition

class Loop(Macro): 
    
    def __init__(self,condition:Condition) -> None:
        super().__init__()     
        self.condition = condition 
        self.child:List[Macro] = []
     
    def toJSON(self):
        return {
            "type": Macro.CONDITION, 
            "condition":  self.condition,
            "child":  self.child
        }  
        
    def label(self):
        return  'Repeat until ' + self.condition.label()
    
    def execute(self):
        pass
